src/main/java/com/redhat/devtools/lsp4ij/usages/LSPExternalReferencesFinder.java

```diff
@@ -0,0 +1,172 @@
+import com.intellij.util.Processor;
+import com.intellij.util.containers.ContainerUtil;
+import com.redhat.devtools.lsp4ij.LSPIJUtils;
+import com.redhat.devtools.lsp4ij.LanguageServerWrapper;
+import com.redhat.devtools.lsp4ij.LanguageServiceAccessor;
+import com.redhat.devtools.lsp4ij.client.features.LSPClientFeatures;
+import com.redhat.devtools.lsp4ij.features.LSPPsiElement;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.*;
+
+/**
+ * Utility class that helps to process/find external references to LSP4IJ-based (pseudo-)elements.
+ */
+public final class LSPExternalReferencesFinder {
+
+    private LSPExternalReferencesFinder() {
+        // Pure utility class
+    }
+
+    /**
+     * Processes all external references for the LSP4IJ element at the offset in the specified file.
+     *
+     * @param file      the file for which the element at the specified offset should be processed for external references
+     * @param offset    the offset of the element in the file
+     * @param processor the external reference processor
+     */
+    public static void processExternalReferences(@NotNull PsiFile file,
+                                                 int offset,
+                                                 @NotNull Processor<PsiReference> processor) {
+        VirtualFile virtualFile = file.getVirtualFile();
+        if (virtualFile != null) {
+            Document document = LSPIJUtils.getDocument(virtualFile);
+            TextRange wordTextRange = document != null ? LSPIJUtils.getWordRangeAt(document, file, offset) : null;
+            if (wordTextRange != null) {
+                LSPPsiElement wordElement = new LSPPsiElement(file, wordTextRange);
+                String wordText = wordElement.getText();
+                if (StringUtil.isNotEmpty(wordText)) {
+                    processExternalReferences(
+                            file,
+                            wordText,
+                            wordTextRange,
+                            ProgressManager.getInstance().getProgressIndicator(),
+                            processor
+                    );
+                }
+            }
+        }
+    }
+
+    private static void processExternalReferences(@NotNull PsiFile file,
+                                                  @NotNull String wordText,
+                                                  @NotNull TextRange wordTextRange,
+                                                  @Nullable ProgressIndicator progressIndicator,
+                                                  @NotNull Processor<PsiReference> processor) {
+        VirtualFile virtualFile = file.getVirtualFile();
+        if (virtualFile == null) {
+            return;
+        }
+
+        Project project = file.getProject();
+        Set<LanguageServerWrapper> languageServerWrappers = LanguageServiceAccessor.getInstance(project).getMatchedLanguageServerWrappersSync(virtualFile);
```

Comment:
Replace with:

```java
boolean caseSensitive =
                LanguageServiceAccessor.getInstance(file.getProject())
                .hasAny(file.getVirtualFile(), ls -> ls.getClientFeatures().isCaseSensitive(file));
```

and remove your new method getMatchedLanguageServerWrappersSync.